specs/kiota.config.md

```diff
@@ -0,0 +1,634 @@
+              "authenticationProvider": "Microsoft.Kiota.Authentication.AzureAuthProvider",
+              "authenticationParameters": {
+                "clientId": "guid"
+              },
+            },*/
+            "serializers": [
+              "Contoso.Json.CustomSerializer"
+            ],
+            "deserializers": [
+              "Contoso.Json.CustomDeserializer"
+            ],
+            "structuredMimeTypes": [
+              "application/json"
+            ],
+            "usesBackingStore": true,
+            "includeAdditionalData": true
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### kiota client delete
+
+`kiota api client` allows a developer to delete an existing client from the `kiota.config` file. The command will remove the entry from the `clients` section of parent API within the `kiota.config` file. The command has two required parameters, the name of the API and the name of the client. The command also has one optional parameter, the ability to remove the generated client. If provided, kiota will delete the folder specified at the `outputPath` from the client configuration.
+
+| Parameters | Required | Example | Description |
+| -- | -- | -- | -- |
+| `--api-name \| --api` | Yes | graph | Name of the API |
+| `--client-name \| --cn` | Yes | graphDelegated | Name of the client |
+| `--clean-output \| --co` | No |  | Cleans the generated client |
+
+#### Using kiota client delete
+
+```bash
+kiota client delete --api-name "graph" --client-name "graphDelegated" --clean-output
+```
+
+```jsonc
+// Removes the following from the kiota.config file
+"graphDelegated": {
+  "language": "csharp",
+  "outputPath": "./generated/graph/csharp",
+  "clientClassName": "GraphClient",
+  "clientNamespaceName": "Contoso.GraphApp",
+  "features": {
+    // Adding for future visibility, but not required for now
+    /*"authentication": {
+      "authenticationProvider": "Microsoft.Kiota.Authentication.AzureAuthProvider",
+      "authenticationParameters": {
+        "clientId": "guid"
+      },
+    },*/
+    "serializers": [
+      "Contoso.Json.CustomSerializer"
+    ],
+    "deserializers": [
+      "Contoso.Json.CustomDeserializer"
+    ],
+    "structuredMimeTypes": [
+      "application/json"
+    ],
+    "usesBackingStore": true,
+    "includeAdditionalData": true
+  }
+}
+```
+
+The resulting `kiota.config` file will look like this:
+
+```javascript
+{
+  "name": "My application",
+  "version": "1.0",
+  "apis": {
+    "graph": {
+      "descriptionHash": "9EDF8506CB74FE44...",
+      "descriptionLocation": "https://.../openapi.yaml",
+      "includePatterns": ["/me/chats#GET", "/me#GET"],
+      "excludePatterns": [],
+      "clients": { }
+    }
+  }
+}
+```
+
+### kiota generate
+
+Now that we have a `kiota.config` file, all the parameters required to generate the code are stored in the file. The `kiota generate` command will read the `kiota.config` file and generate the code for each of the clients. 
```

Comment:
If we want to keep the current CLI surface compatibility (not saying that we absolutely have to), we might want to keep the generate command as it. And when you call it, it'd do the following:
1. init (if required)
2. add (if required)
3. update